File: monitoring/src/main/java/com/example/monitoring/user/application/port/in/response/UserResponseDto.java
package com.example.monitoring.user.application.port.in.response;


import com.example.monitoring.user.domain.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponseDto {

    private Long id;
    private String name;
    private int age;

    public static UserResponseDto from(User user) {
        return UserResponseDto.builder()
                .id(user.getId())
                .name(user.getName())
                .age(user.getAge())
                .build();
    }
}

File: kotlin-spring-boot/src/main/kotlin/com/hello/kotlin/user/application/response/UserResponseDto.kt
package com.hello.kotlin.user.application.response

import com.hello.kotlin.user.domain.User

data class UserResponseDto private constructor(
    val id: Long,
    val name: String,
    val age: Int,
    val address: String,
) {
    companion object {
        fun of(user: User) = UserResponseDto(
            id = user.id!!,
            name = user.name,
            age = user.age,
            address = user.address
        )
    }
}
